Meetup 23 resources (#181)
w3c/wot-cg: Meetups/23/2025-02-20-minutes.md

### Question

Ege: could you explain your process of getting to something working?

Andreas: the platform is data driven. We start by loading data. You can create tables very easily once you have data in the DB. Everything has a page (much like a semantic Wikipedia). You can customize any page. Everything can be committed to github and personalized.

Ege: Which opensource tools did you use? What are the tools and libraries that make it click?

Andreas: thanks to standards and JSON schema.

Henk: What discovery mechanism did you use? Very impressing presentation, and I can tell there were a lot of work.

Andreas: To discover a new thing I will just input the Thing Description URL. In terms of the protocols we only support the easy stuff URL and JSON.

Henk: Are you planning any "automatic loading" of TDs?

Andreas: we have a registry CSV or a north bound API.

Andreas: there are devices protected by password. We added a small extension to TD to express the role and the accounts who can use the device.

Henk: If you have 5 people with different roles, do they access to different part of the system? or they can't do some actions?

> Andreas: on the database level there is a table which contains all the things and two other tables connected with Properties and Actions. We can lock different part of this table to different roles. there for we can support hiding or reviling Properties or Actions you can view and use.
> 
> Andreas: we are interested working with the community and love to hear more from you. Please use our github repository. The project is at his beginning.
> 
> Ege: any inputs for the standardization process?

Reviewer: ```suggestion
Andreas: On the database level, there is a table which contains all the things, and two other tables connected with Properties and Actions. We can lock different parts of these tables to different roles. We can support hiding or revealing Properties or Actions you can view and use.

Andreas: We are interested working with the community and love to hear more from you. Please use our GitHub repository. The project is just beginning!

Ege: Any inputs for the standardization process?
```

> Andreas: we are still working of some of the feature. But from the research perspective is semantic harmonization. It would be wonderful to use JSON-LD and semantic more to automate this process.
> 
> Vignesh: We there is something that we need to used but is not in the standard is it okay to be added in the TD?
> 
> Andreas: we ignore them, but they don't show in the UI. You can make extension of the WoT manager to use this information.

Reviewer: ```suggestion
Andreas: We are still working on some of the features. But from the research perspective is semantic harmonization. It would be wonderful to use JSON-LD and semantic more to automate this process.

Vignesh: When there is something that we need to use but it is not in the standard, is it okay to be added in the TD?

Andreas: We ignore them, so they don't show in the UI. You can make an extension of the WoT manager to use this information.
```

> Cris: are you storing timeseries in the postgres database
> 
> Andreas: we are storing the latest value, but it is easy to change.
> 
> Cris: is it polling based?
> 
> Andreas: yes.

Reviewer: ```suggestion
Cris: Are you storing timeseries in the PostgreSQL database?

Andreas: We are storing the latest value, but it is easy to change.

Cris: Is it polling based?

Andreas: Yes.
```

> Ege: Interesting point. Describing which properties needs to be stored in a timeseries db could be something we can do in the specification.
> 
> Andreas: whether to keep the value or not is definitely something not device specific.
> 
> Andreas: what do think about the AI agent? For me the interesting part is that you don't have to think about different application context.
> 
> Ege: it is interesting that you showed how you can achieve a good interaction with the AI with off the shelf tools without reinviting any AI specific. Probably, in this case you can use quite small AI models to achieve good results.

Reviewer: ```suggestion
Ege: Interesting point. Describing which properties need to be stored in a timeseries DB could be something we can do in the specification.

Andreas: Whether to keep the value or not is definitely not device specific.

Andreas: What do you think about the AI agent? For me, the interesting part is that you don't have to think about different application context.

Ege: It is interesting that you showed how you can achieve a good interaction with the AI with off-the-shelf tools without reinventing any AI specification. In this case, you can probably use quite small AI models to achieve good results.
```

> Andreas: we pre process the TD before send it to the LLM.
> 
> Cris: what are the limitations?
> 
> Andreas: in complex contexts it gets messy.
> 
> Cris: are you polishing the AI output?

Reviewer: ```suggestion
Andreas: We pre-process the TD before sending it to the LLM.

Cris: What are the limitations?

Andreas: In complex contexts, it gets messy.

Cris: Are you polishing the AI output?
```

> Andreas: currently is directly connected.
> 
> Cris: do you have some Role access control for AI agents?
> 
> Andreas: User privileges are in there but they use a different key.
> 
> Ege: Using AI prompts used in same way can help solving the data augmentation problem?
> 
> Andreas: not really, the data mapping should be very precise. But it could be a good idea.

Reviewer: ```suggestion
Andreas: Currently it is directly connected.

Cris: Do you have some Role-based access control for AI agents?

Andreas: User privileges are in there, but they use a different key.

Ege: Can using AI prompts help solve the data augmentation problem?

Andreas: Not really, the data mapping should be very precise. But it could be a good idea.
```
